consensus/validator.go

```diff
@@ -140,13 +142,11 @@ func (consensus *Consensus) onPrepared(msg *msg_pb.Message) {
 		consensus.getLogger().Error().Err(err).Msg("ReadSignatureBitmapPayload failed!")
 		return
 	}
-
 	if !consensus.Decider.IsQuorumAchievedByMask(mask) {
-		consensus.getLogger().Warn().
-			Msgf("[OnPrepared] Quorum Not achieved")
+		consensus.getLogger().Warn().Msgf("[OnPrepared] Quorum Not achieved.")
+		consensus.spinUpStateSync()
```

Comment:
not sure this logic is correct. If the quorum check failure is caused by node out of sync, they it should detect that by checking block number somewhere else.

Reply:
This is the temporary hack before we add the leader public key check. Considering the attack scenario, we can only spin up state sync if 

1. block number is left behind. 
2. Message is sent by leader. 
3. Message passed all validation (requires latest block in state and consensus cache).

Currently, the second part is missing and state sync is spinned up after all check pass. Only if we have leader public key verified (in the next PR), we can spin up state sync in condition 1+2 or 2+3.

Reply:
The leader check is outside `onCommitted` and `onPrepared`. Added spin up sync logic right after low block number is detected.